### PR TITLE
Feature has_many_aggregate on a through relationship.

### PR DESCRIPTION
**Pull request by kadenbarlow**
This adds support for has_many_aggregate on through relationships. Currently, the SQL generated is invalid because the query is missing a join and some conditionals. The specs that were added fail on the master branch but pass with the new functionality.

### CODE REVIEW

File: lib/jit_preloader/active_record/base.rb [hunk 1]
```diff
@@ -41,17 +41,39 @@ def has_many_aggregate(assoc, name, aggregate, field, default: 0)
...
+            reflections = []
+            while through do
+              through_reflection = source_class.reflect_on_association(through)
+              reflections << through_reflection
+              through = through_reflection.options[:through]
+            end
+
+            conditions["#{source_class.table_name}.#{source_class.primary_key}"] = primary_ids
+
+            join_for_group_by = "INNER JOIN #{source_class.table_name} on #{reflections.last.table_name}.#{reflections.last.foreign_key} = #{source_class.table_name}.#{source_class.primary_key}"
```
Comment (doliveirakn) [inline lib/jit_preloader/active_record/base.rb]: This code here is fairly dense, and hard to follow. 

We are also rebuilding some internal Rails logic to construct the joins and I am not 100% confident that it covers all of the edge cases. (Though your specs have been a great help!)

I think there is a way we can leverage the Rails internals to do this work for us, and at the same time, not worry about having to have a large branch on the `through`

Comment (kadenbarlow) [reply lib/jit_preloader/active_record/base.rb]: @doliveirakn Yes, this looks great! I like that there isn't a separate if branch for the through relationships, looks much cleaner now. I really like the idea of ActiveRecord constructing the query and not doing it manually, I couldn't quite figure out how to do that before but I'm glad that you looked through and did this.  

I went ahead and modified this PR. The only thing I changed in your code example was the naming of first_association to aggregate_association which I thought was a little easier to follow but really wasn't a big deal if you want me to change that back I can.

**Comment by doliveirakn** [comment]
@kadenbarlow This has been merged, and released as `0.2.1`. Thank you for your contribution.
